processors/src/main/java/io/opentelemetry/contrib/eventbridge/EventToSpanEventBridge.java

```diff
@@ -0,0 +1,130 @@
+import io.opentelemetry.sdk.logs.ReadWriteLogRecord;
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A processor that records events (i.e. log records with an {@code event.name} attribute) as span
+ * events for the current span if:
+ *
+ * <ul>
+ *   <li>The log record has a valid span context
+ *   <li>{@link Span#current()} returns a span where {@link Span#isRecording()} is true
+ * </ul>
+ *
+ * <p>The event {@link LogRecordData} is converted to attributes on the span event as follows:
+ *
+ * <ul>
+ *   <li>{@code event.name} attribute is mapped to span event name
+ *   <li>{@link LogRecordData#getTimestampEpochNanos()} is mapped to span event timestamp
+ *   <li>{@link LogRecordData#getAttributes()} are mapped to span event attributes, excluding {@code
+ *       event.name}
+ *   <li>{@link LogRecordData#getObservedTimestampEpochNanos()} is mapped to span event attribute
+ *       with key {@code log.record.observed_timestamp}
+ *   <li>{@link LogRecordData#getSeverity()} is mapped to span event attribute with key {@code
+ *       log.record.severity_number}
+ *   <li>{@link LogRecordData#getBodyValue()} is mapped to span event attribute with key {@code
+ *       log.record.body}, as an escaped JSON string following the standard protobuf JSON encoding
+ *   <li>{@link LogRecordData#getTotalAttributeCount()} - {@link
+ *       LogRecordData#getAttributes()}.size() is mapped to span event attribute with key {@code
+ *       log.record.dropped_attributes_count}
+ * </ul>
+ */
+public final class EventToSpanEventBridge implements LogRecordProcessor {
+
+  private static final Logger LOGGER = Logger.getLogger(EventToSpanEventBridge.class.getName());
+
+  private static final AttributeKey<String> EVENT_NAME = AttributeKey.stringKey("event.name");
+  private static final AttributeKey<Long> LOG_RECORD_OBSERVED_TIME_UNIX_NANO =
+      AttributeKey.longKey("log.record.observed_time_unix_nano");
+  private static final AttributeKey<Long> LOG_RECORD_SEVERITY_NUMBER =
+      AttributeKey.longKey("log.record.severity_number");
+  private static final AttributeKey<String> LOG_RECORD_BODY =
+      AttributeKey.stringKey("log.record.body");
+  private static final AttributeKey<Long> LOG_RECORD_DROPPED_ATTRIBUTES_COUNT =
+      AttributeKey.longKey("log.record.dropped_attributes_count");
+
+  private EventToSpanEventBridge() {}
+
+  /** Create an instance. */
+  public static EventToSpanEventBridge create() {
+    return new EventToSpanEventBridge();
+  }
+
+  @Override
+  public void onEmit(Context context, ReadWriteLogRecord logRecord) {
+    LogRecordData logRecordData = logRecord.toLogRecordData();
+    String eventName = logRecordData.getAttributes().get(EVENT_NAME);
+    if (eventName == null) {
+      return;
+    }
```

Comment:
It is disappointing and needlessly wasteful. We have a [todo](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/logs/src/main/java/io/opentelemetry/sdk/logs/ReadWriteLogRecord.java#L50-L51) to add more getters for directly accessing bits of info in `ReadWriteLogRecord`. May be time to pick that up..

Separately, I wonder if we can / should add a helper method `boolean isEvent()` since we anticipate this being a common thing to access.

Reply:
Yeah, I like that idea. We can revisit this here later when that happens.

Reply:
Opened this issue to track: https://github.com/open-telemetry/opentelemetry-java/issues/6895